loader.php

```diff
@@ -15,45 +15,117 @@ function is_development() {
 }
 
 /**
- * Get the port for React's development server.
+ * Attempt to load a file at the specified path and parse its contents as JSON.
  *
- * @return int|null Port number if available, otherwise null.
+ * @param string $path The path to the JSON file to load.
+ * @return array|null;
  */
-function get_react_port() {
-	if ( ! is_development() ) {
+function load_asset_file( $path ) {
+	if ( ! file_exists( $path ) ) {
 		return null;
 	}
-
-	$path = get_theme_file_path( 'react-port' );
-	if ( ! file_exists( $path ) ) {
+	$contents = file_get_contents( $path );
+	if ( empty( $path ) ) {
 		return null;
 	}
+	return json_decode( $contents, true );
+}
 
-	$port = file_get_contents( $path );
-	if ( empty( $port ) || ! is_numeric( $port ) ) {
-		return null;
+/**
+ * Check a directory for a root or build asset manifest file, and attempt to
+ * decode and return the asset list JSON if found.
+ *
+ * @param string $directory Root directory containing `src` and `build` directory.
+ * @return array|null;
+ */
+function get_assets_list( string $directory ) {
+	$directory = trailingslashit( $directory );
```

Comment:
I'd rather `untrailingslashit`, I generally prefer `$x . '/filename'` over `$x . 'filename'`, I think it reads a bit better.

Reply:
The manifest file we use elsewhere contains paths without leading slashes, so it felt more appropriate to consistently have trailing-slash'd base strings and paths than to have it be situational. I'll change if you feel like it makes a difference but this feels more consistent to me..